Dockerfile

```diff
@@ -37,14 +46,88 @@ RUN git clone --depth=1 --single-branch --branch=frankenphp-8.2 https://github.c
         --enable-embed=static \
         --enable-zts \
         --disable-zend-signals \
-        --with-config-file-scan-dir=/usr/local/lib/php/ && \
+    	# --enable-mysqlnd is included here because it's harder to compile after the fact than extensions are (since it's a plugin for several extensions, not an extension in itself)
+    	--enable-mysqlnd \
+     	# make sure invalid --configure-flags are fatal errors instead of just warnings
+    	--enable-option-checking=fatal \
+    	# https://github.com/docker-library/php/issues/439
+    	--with-mhash \
+    	# https://github.com/docker-library/php/issues/822
+    	--with-pic \
+    	# --enable-ftp is included here because ftp_ssl_connect() needs ftp to be compiled statically (see https://github.com/docker-library/php/issues/236)
+    	--enable-ftp \
+    	# --enable-mbstring is included here because otherwise there's no way to get pecl to use it properly (see https://github.com/docker-library/php/issues/195)
+    	--enable-mbstring \
+    	# https://wiki.php.net/rfc/argon2_password_hash
+    	--with-password-argon2 \
+    	# https://wiki.php.net/rfc/libsodium
+    	--with-sodium=shared \
+    	# always build against system sqlite3 (https://github.com/php/php-src/commit/6083a387a81dbbd66d6316a3a12a63f06d5f7109)
+		--with-pdo-sqlite=/usr \
+		--with-sqlite3=/usr \
+		--with-curl \
+		--with-iconv \
+		--with-openssl \
+		--with-readline \
+		--with-zlib \
+    	# https://github.com/bwoebi/phpdbg-docs/issues/1#issuecomment-163872806 ("phpdbg is primarily a CLI debugger, and is not suitable for debugging an fpm stack.")
+		--disable-phpdbg \
+        --with-config-file-path="$PHP_INI_DIR" \
+        --with-config-file-scan-dir="$PHP_INI_DIR/conf.d" && \
     make -j$(nproc) && \
     make install && \
     rm -Rf php-src/ && \
     ldconfig && \
+    install-php-extensions opcache && \
     php --version
 
-RUN echo "zend_extension=opcache.so\nopcache.enable=1" > /usr/local/lib/php.ini
+#RUN echo "zend_extension=opcache.so\nopcache.enable=1" > /usr/local/lib/php.ini
+
+FROM golang AS server
+
+ENV PHPIZE_DEPS \
+    autoconf \
+    dpkg-dev \
+    file \
+    g++ \
+    gcc \
+    libc-dev \
+    make \
+    pkg-config \
+    re2c
+
+RUN apt-get update && \
+    apt-get -y --no-install-recommends install \
+    $PHPIZE_DEPS \
+    libargon2-dev \
+    libcurl4-openssl-dev \
+    libonig-dev \
+    libreadline-dev \
+    libsodium-dev \
+    libsqlite3-dev \
+    libssl-dev \
+    libxml2-dev \
+    zlib1g-dev \
+    bison \
+    && \
+    apt-get clean
+
+RUN git clone --depth=1 --single-branch --branch=frankenphp-8.2 https://github.com/dunglas/php-src.git && \
```

Comment:
We should be able to copy `libphp.a` and the headers build in the previous stage instead of rebuilding again.

We just need to ensure that we use bullseye here too.

Reply:
Now the build is failing:
```
#30 22.78 /usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
#30 22.78 /usr/bin/ld: /usr/local/lib/libphp.a(zlib.o): undefined reference to symbol 'inflateEnd'
#30 22.78 /usr/bin/ld: /lib/x86_64-linux-gnu/libz.so.1: error adding symbols: DSO missing from command line
#30 22.78 collect2: error: ld returned 1 exit status
```

My knowledge of the intimate details of the  build environment is very low, so I'm not sure what this means. Any ideas?

Reply:
Ok, we also need to install/copy all C libraries installed in the PHP image in the builder image. It's mandatory to build the Go binary. Here it's zlib, but it will be the same for most/all libraries installed in the PHP image.